relay-common/src/constants.rs

```diff
@@ -96,6 +96,10 @@ impl fmt::Display for EventType {
 #[repr(i8)]
 pub enum DataCategory {
     /// Reserved and unused.
+    ///
+    /// There is a reserved `internal` data category which also maps to `Default`.  Relays
```

Comment:
Why do we need this? From what I understand we never emit this within an outcome nor send it as rate-limit as you said. All SDKs apparently need to know about is the item type.

Reply:
I wrote the reasoning on this into the SDK docs. Copying here for reference:

------------

Special note on the internal data category: this data category exists but is promised not to be emitted as we do not enforcd
rate limits on internal items.  This might appear odd but the purpose of this is to allow envelope implementations in SDKs to
assign a data category to internal items such as `client_report`.  SDKs can handle `internal` like any other data category but
they can rely on the fact that rate limits are not communicated for these explicitly.  That said, the special _all_ category
does apply to `internal` event types as well.

Reply:
Introduced an explicit non propagated internal data category.